src/components/ReturnsCalculator.jsx

```diff
@@ -0,0 +1,244 @@
+
+const textColor = { light: "gray.600", dark: "#FFF" };
+
+type ReturnsCalculatorProps = {
+	colorMode: "light" | "dark",
+	validatorData: Array<{}>,
+};
+
+const ReturnsCalculator = (props: ReturnsCalculatorProps) => {
+	const { colorMode, toggleColorMode }        = useColorMode();
+	const [stakeInput, setStakeInput]           = React.useState();
+	const [expectedReturns, setExpectedReturns] = React.useState(0.0);
+	const [suggPromptsAmount]                   = useDebounce(stakeInput / 16, 0);
+	const [validatorsList, setValidatorsList]   = React.useState([]);
+	const [validatorData, setValidatorData]     = React.useState([]);
+	const [errorState, setErrorState]           = React.useState(false);
+	const [intentionData, setIntentionData]     = React.useState([]);
+	const [apiConnected, setApiConnected]       = React.useState(false);
+	const [isLoaded, setIsLoaded]               = React.useState(false);
+	const [riskLevel, setRiskLevel]             = React.useState(50);
+	const [sliderBG, setSliderBG]               = React.useState("yellow.300");
+	const ERA_PER_DAY = 4;
+	// console.log('props - ', props.validatorData);
+
+	function suggPrompts() {
+		const data = validatorsList.map(validator => {
+			const {
+				stashId,
+				stashIdTruncated,
+				name,
+				commission,
+				totalStake,
+				poolReward,
+				noOfNominators
+			} = validator;
+			const userStakeFraction =
+				suggPromptsAmount / (suggPromptsAmount + totalStake);
+			const dailyEarning = userStakeFraction * poolReward * ERA_PER_DAY;
+			return {
+				noOfNominators,
+				stashId,
+				stashIdTruncated,
+				name,
+				commission: `${parseFloat(commission)}%`,
+				dailyEarning: isNaN(dailyEarning)
+					? "Not enough data"
+					: `${dailyEarning.toPrecision(10)} KSM`,
+				dailyEarningPrecise: isNaN(dailyEarning) ? 0 : dailyEarning
+			};
+		});
+
+		//handle top 16 suggested validaors
+		data.sort((a, b) => b.dailyEarningPrecise - a.dailyEarningPrecise);
+		const top16data = [...data.slice(0, 16)];
+		//console.log("table data of top 16 val - ", top16data);
+		
+		//pass props to parent App.jsx
+		const propsData = {};
+		if (top16data.length > 0) {
+			// eslint-disable-next-line no-unused-vars
+			const expectedEarning = top16data.reduce((a, b) => ({
+				dailyEarningPrecise: a.dailyEarningPrecise + b.dailyEarningPrecise
+			}));
+			//console.log("expected earning of top 16 val - ", expectedEarning);
+			propsData.expectedReturns = expectedEarning.dailyEarningPrecise;
+			setExpectedReturns(expectedEarning.dailyEarningPrecise);
+		}
+		propsData.validatorsList = top16data;
+		propsData.budget = stakeInput;
+		props.onEvent(propsData);
+		
+		if (apiConnected) setIsLoaded(true);
+	}
+
+	React.useEffect(() => {
+		setValidatorsList(props.validatorData);
+	}, [props]);
+
+	if (errorState) {
+		return <ErrorMessage />;
+	}
+
+	function calculate() {
+		suggPrompts();
+	}
+
+	const onRiskChange = value => {
+		console.log(value);
+		if (value <= 30) {
```

Comment:
Use the constants for `LOW_RISK`, `MEDIUM_RISK`, `HIGH_RISK` defined by Abhinav in his PR, you might need to merge develop locally

Reply:
@saumyakaran Can you please explain this?

Reply:
Refer https://github.com/buidl-labs/YieldScan/blob/db49c4a27f36c6dbd972412de42ff11000b954f5/src/constants.js

Reply:
different function is added for this because green, red, yellow doesn't work here.

Reply:
> different function is added for this because green, red, yellow doesn't work here.

I don't see how this is relevant. Why can't we replace hard-coded "30" with a global variable called `LOW_RISK`?